From 2adb71dc192394733de9ea2efc50483abf05d1eb Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 11 Jul 2022 14:51:50 +0300
Subject: [PATCH] [SW94985]: Revert "[SW-94323] gaudi2: fix bug in clearing recovery flag."

This reverts commit 62ef2e7c2395b7ad96fd872ac3f5e5d2ca8770f8.
Since the recovery flag is cleared in erom_init if the firmware update
fails, the arcboot rom will try to load ppboot instead of executing the
erom recovery. Therefore the recovery flag should be cleared only after
the successful firmware update i.e, in zephyr agent and hl-fw-loader.

Change-Id: I62309d0f36af61e9fd5e3c4ec89a5dd362b89462
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 4547e4f..d3aed0c 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1094,7 +1094,7 @@
 	struct gaudi2_cold_rst_data cold_rst_data;
 
 	cold_rst_data.data = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-	cold_rst_data.recovery_flag = 0;
+	cold_rst_data.recovery_flag = 1;
 	regmap_writel(cold_rst_data.data, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
 }
 
